### ISSUE TITLE
Grad clipping for fp16

### ISSUE THREAD

**Issue by nnethercott** [kaistAI/prometheus#10]
Just a question concerning `train` in train/utils/train_utils.py, but I was curious why gradient clipping is used in the full precision training but not in half precision (comparing code contained in lines 97-113).  Is there a specific reason the code is setup this way? Or in other words why no `scaler.unscale_(optimizer)` followed by `torch.nn.utils.clip_grad_norm_(model.parameters(), 0.8)`  when `train_config.use_fp16=True` ?

**Comment by SeungoneKim** [comment]
@nnethercott Hello Nate, thank you for your interest in our paper, and sorry for the late response!

We only trained with bf16, so you could ignore the code for fp16! I've tweaked the gradient clipping value multiple times (including my previous work) and empirically 0.7~0.8 seemed to work well when generating Chain-of-Thoughts (the language feedback in this project)!

Hope this helps!
